controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1057,106 +1058,53 @@ func (r *NnfWorkflowReconciler) createOrUpdateContainerJobsIfNecessary(ctx conte
 			},
 		}
 
-		// Apply Job Labels
-		// TODO: confirm labels here; any missing?
+		// Apply Job Labels/Owners
```

Comment:
Optimizations:
* 99% of the job spec is the same across all jobs. You should create one structure up front, apply labels, volumes, etc. I think only the nodeSelector actually needs to be modified inside the loop.
* Watch your logs inside these loops: You may see only 1 log today, but that'll be hundreds in production.

Reply:
You're right. Only job `name`, `hostname`, and `nodeSelector` need to be changed. I've optimized the job creation loop to just share 1 job and then update those 3 fields for every NNF node (since they are based on the NNF Node name).